src/main/java/net/minecraftforge/event/level/AlterGroundEvent.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.event.level;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.util.RandomSource;
+import net.minecraft.world.level.LevelSimulatedReader;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.levelgen.feature.treedecorators.TreeDecorator;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.eventbus.api.Cancelable;
+import net.minecraftforge.eventbus.api.Event;
+
+/**
+ * This event is fired when {@link net.minecraft.world.level.levelgen.feature.treedecorators.AlterGroundDecorator#placeBlockAt(TreeDecorator.Context, BlockPos)}
+ * attempts to alter a ground block when generating a feature. An example of this would be large spruce trees converting grass blocks into podzol.
+ * <p>
+ * This event is {@linkplain Cancelable cancellable}.
```

Comment:
What does canceling do in this case? From what I understand of the original implementation, it just doesn't change the block.